plugins/synapticsmst/fu-plugin-synapticsmst.c

```diff
@@ -39,18 +37,108 @@ synapticsmst_common_check_supported_system (FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
+/* creates MST-$str-$BOARDID */
+static void
+fu_plugin_synapticsmst_create_simple_guid (FuDevice *fu_device,
+					   SynapticsMSTDevice *device,
+					   const gchar *str)
+{
+	guint32 board_id = synapticsmst_device_get_board_id (device);
+	g_autofree gchar *guid = g_strdup_printf ("MST-%s-%u", str, board_id);
+
+	fu_device_add_guid (fu_device, guid);
+}
+
+/* creates MST-$str-$chipid-$BOARDID */
+static void
+fu_plugin_synapticsmst_create_complex_guid (FuDevice *fu_device,
+					    SynapticsMSTDevice *device,
+					    const gchar *str)
```

Comment:
`str` doesn't give me the warm-and-fuzzies -- what actually is this?

Reply:
In it's current use it's the model of dock, eg "wd15", "tb16". You can see the contents from the `synapticsmst.quirk` file.

Reply:
Maybe `dock_kind` then?

Reply:
Well I use it as `dock_kind`, but some day maybe that doesn't fit with what the string going into here is.